railties/test/commands/server_test.rb

```diff
@@ -122,11 +122,11 @@ def test_log_stdout
   end
 
   def test_default_options
-    old_default_options = parse_arguments
+    server = Rails::Server.new
+    old_default_options = server.default_options
 
     Dir.chdir("..") do
-      default_options = parse_arguments
-      assert_equal old_default_options, default_options
+      assert_equal old_default_options, server.default_options
```

Comment:
The value of `Rails::Command::ServerCommand#server_options` is changed when path is changed. Because it made it to acquire `pid` path dynamically. 
This is because the directory is changed according to `APP_PATH` at server startup, so it is not possible to use a fixed path at the time of file loading. 
https://github.com/rails/rails/blob/master/railties/lib/rails/command/actions.rb#L8

Therefore, it is correct that `Rails::Command::ServerCommand#server_options` will change when the directory changes, so I think the original test is incorrect.  
However, the purpose of the test was first added, it was to confirm that `Rails::Server#default_options` returns the same result. So I changed to use `Rails::Server#default_options`.
Ref: 221b4ae